### PR TITLE
feat(cli): add TES support

### PR DESCRIPTION
**Pull request by aaravm**
This PR contains code for the CLI and Instructions on how to use them. It is inspired from https://github.com/vemonet/nanopub-rs/blob/main/cli/src/main.rs

## Summary by Sourcery

Add a new CLI tool for TES task management with commands for creating, listing, retrieving, checking the status of, and canceling tasks, along with usage documentation.

New Features:
- Introduce a new CLI tool for managing TES tasks, including commands for creating, listing, retrieving, checking the status of, and canceling tasks.

Documentation:
- Add usage examples and instructions for the new CLI tool in the main.rs file.

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+///
+/// ```sh
+/// cargo run -- tes cancel cqgk5lj93m0311u6p530      
+/// ```
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    let cmd = Command::new("cli")
+        .bin_name("cli")
+        .version("1.0")
```
Comment (pavelnikonorov) [inline cli/src/main.rs:73]: i think more like 0.1.0

**Review by pavelnikonorov — CHANGES_REQUESTED** [review]
I've added a few minor suggestions, and one regarding the list command interface, which you might decide to formulate as an issue or improve within the current PR.

File: cli/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+///         "command": ["cat", "/inputs/hello.txt"],
+///         "stdout": "/outputs/stdout"
+///     }]
+/// }'
+/// ```
+///
+/// Or:
+///
+/// ```sh
+/// cargo run -- tes create './tests/sample.tes'
```
[the remark this answers is not in the document]
Comment (aaravm) [reply cli/src/main.rs]: @pavelnikonorov I am not quite sure what you mean by commas in the path. Could you please elaborate?

[the remark this answers is not in the document]
Comment (aaravm) [reply cli/src/main.rs]: If you mean whether it can run like:
```
cargo run -- tes create ./tests/sample.tes
```
then, yes it can. I will update it

**Comment by aaravm** [comment]
@pavelnikonorov, I have addressed all of your concerns. Pleas let me know if you think any other changes should be made.

**Comment by uniqueg** [comment]
@aaravm: Can you please merge #37 (after updating the PR description) and then update this PR so that it only contains the changes relative to the updated `main`? I will give it a closer review then.

For now: Please make sure all modules have docstrings for the module, class, function level etc., as well as comprehensive doctests. Same thing as for the other PRs...

**Comment by aaravm** [comment]
Hi @uniqueg @pavelnikonorov, I have changed the list CLI, to take input in the form of 
```
ga4gh-cli tes list --name_prefix <NAME_PREFIX> --state <STATE> --tag_key <TAG_KEY> --tag_value <TAG_VALUE> --page_size <PAGE_SIZE> --page_token <PAGE_TOKEN> --view <VIEW>
```
instead of dictionary, as suggested by @pavelnikonorov . Please look into this PR and merge it

**Review by pavelnikonorov — APPROVED** [review]
@uniqueg I've tested the CLI; improved the task list output and created an issue for one edge case https://github.com/elixir-cloud-aai/ga4gh-sdk/issues/46.

**Comment by pavelnikonorov** [comment]
all suggestions addressed, merging as discussed with @uniqueg

**Pull request merged**